std/bitmanip.d

```diff
@@ -1570,9 +1828,13 @@ public:
     }
 
     /***************************************
-     * Support for unary operator ~ for $(D BitArray).
+     * Support for unary operator `~` for `BitArray`. This is the
+     * equivalent of `flip` which does not modify the original array.
+     *
+     * Returns: Copy of the `BitArray` with all bits flipped.
      */
-    BitArray opCom() const pure nothrow
+    BitArray opUnary(string op)() const pure nothrow
```

Comment:
These operator functions all allocate memory for the return value. While pedantically correct, they are a disaster to use. Any significant math will put a lot of pressure on the GC, making it agonizingly slow. The whole point of using bit vectors is for speed, and these functions just don't deliver on that.

They need to have another parameter which is where to place the return value. Of course, then operator overloading won't work anymore. The operator overloading functions should be deprecated.

Reply:
See src/dmd/backend/tk/vec.c